app/renderer/src/containers/AppContainer.js

```diff
@@ -29,19 +39,42 @@ const AppContainer = class extends Component {
     }
   }
 
+  updateQuery = q => {
+    this.setState({ q });
+    ipcRenderer.send(IPC_QUERY_COMMAND, { q });
+  };
+
+  resetQuery = () => {
+    // @todo - just clearn the results once results actions/reducers are removed
+    this.props.resetResults && this.props.resetResults();
+  };
+
   render() {
     const { theme } = this.props;
-    return <App theme={theme} />;
+    const { q } = this.state;
+
+    return (
+      <App
+        q={q}
+        theme={theme}
+        onQueryChange={this.updateQuery}
+        onQueryReset={this.resetQuery}
+      />
+    );
   }
 };
 
 AppContainer.defaultProps = {
   theme: {},
+
+  // redux-actions
   setTheme: () => {},
 };
 
 AppContainer.propTypes = {
   theme: ThemeSchema,
+
+  // redux-actions
   setTheme: PropTypes.func,
```

Comment:
hoh... at this stage, this is actually a `setState`, but let's deal with that later